app/src/main/java/com/example/hw_catsapi/ui/breed/CatsBreedFragment.kt

```diff
@@ -0,0 +1,78 @@
+package com.example.hw_catsapi.ui.breed
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.hw_catsapi.CatsApplication
+import com.example.hw_catsapi.R
+import com.example.hw_catsapi.adapter.CatsBreedAdapter
+import com.example.hw_catsapi.databinding.FragmentCatsBreedBinding
+import com.example.hw_catsapi.repository.Loading
+import com.example.hw_catsapi.utils.addBottomSpaceDecorationRes
+import com.example.hw_catsapi.utils.addPagingScrollListener
+
+class CatsBreedFragment : Fragment() {
+
+    private var _binding: FragmentCatsBreedBinding? = null
+    private val binding get() = requireNotNull(_binding) { "binding is null $_binding" }
+    private val viewModel by viewModels<CatsBreedViewModel> {
+        CatsBreedViewModelFactory((requireActivity().application as CatsApplication).repository)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return FragmentCatsBreedBinding.inflate(layoutInflater, container, false)
+            .also { _binding = it }
+            .root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val catsAdapter = CatsBreedAdapter(requireContext()) {
+            findNavController().navigate(CatsBreedFragmentDirections.toDescription(it))
+        }
+
+        with(binding) {
+
+
+            swipeRefresh.setOnRefreshListener {
+                viewModel.refresh()
+                viewModel.getLoadingStatus().observe(viewLifecycleOwner) {
+                    if (it == Loading.NOT_LOADING) {
+                        swipeRefresh.isRefreshing = false
+                    }
+                }
+            }
+
+            recycler.apply {
+                val layout = LinearLayoutManager(requireContext())
+                adapter = catsAdapter
+                layoutManager = layout
+                addBottomSpaceDecorationRes(resources.getDimensionPixelSize(R.dimen.item_space_bottom))
+                addPagingScrollListener(layout, 10) {
+                    viewModel.fetchBreeds().observe(viewLifecycleOwner) {
+                        catsAdapter.submitList(it.toList())
+                    }
+                }
```

Comment:
здесь опять же, подписку лучше делать не в колбеке, а делать подписку именно в onViewCreated, а в данном методе просто вызывать что-то вроде такого
```suggestion
                addPagingScrollListener(layout, 10) {
                    viewModel.loadNextPage()
                }
```